### PR TITLE
Add `NullFastlyCache` service, use it in development

### PR DESCRIPTION
**Pull request by di**
This adds a origin cache service that mimics our Fastly service in local development, and outputs the requests it would issue in production.

### CODE REVIEW

File: warehouse/cache/origin/fastly.py [hunk 1]
```diff
@@ -143,3 +143,23 @@ def purge_key(self, key, metrics=None):
...
+        url = urllib.parse.urljoin(self.api_endpoint, path)
+        headers = {
+            "Accept": "application/json",
+            "Fastly-Key": self.api_key,
+            "Fastly-Soft-Purge": "1",
+        }
+
+        print("Origin cache purge issued:")
+        print(f"* URL: {url!r}")
+        print(f"* Headers: {headers!r}")
```
Comment (miketheman) [inline warehouse/cache/origin/fastly.py:165]: **question:** ‏should we use print or logging?

**Review by miketheman — APPROVED** [review]
Looks good - a couple of non-blockers, so you choose if you want.

File: tests/unit/cache/origin/test_fastly.py [hunk 2]
```diff
@@ -610,3 +610,28 @@ def test_purge_key_no_fallback(
...
+        )
+        cacher = fastly.NullFastlyCache.create_service(None, request)
+        cacher.purge_key("one", metrics=metrics)
+
+        captured = capsys.readouterr()
+        expected = """
+Origin cache purge issued:
+* URL: 'https://api.example.com/service/the service id/purge/one'
+* Headers: {'Accept': 'application/json', 'Fastly-Key': 'the api key', 'Fastly-Soft-Purge': '1'}
+"""  # noqa
```
[the remark this answers is not in the document]
Comment (di) [reply tests/unit/cache/origin/test_fastly.py:636]: The `noqa` here is because line 635 is too long

File: warehouse/cache/origin/fastly.py (hunk 1 above)
Comment (di) [reply warehouse/cache/origin/fastly.py:165]: Given that we use `print` in other `Null`-services in local dev, I think this is probably fine, not sure if we'd get anything by using logging.

**Pull request merged**